### PR TITLE
Add a find_links method to the browser 

### PR DESCRIPTION
**Pull request by aruznieto**
Find all links matching the given regular expression pattern.

### CODE REVIEW

File: twill/utils.py [hunk 1]
```diff
@@ -128,6 +128,17 @@ def find_link(self, pattern: str) -> Optional[Link]:
...
         return None
+    
+    def find_links(self, pattern: str) -> Optional[List[Link]]:
+        """Find all links with a given pattern on the result page."""
+        regex = re.compile(pattern)
+        links = []
+        for link in self.links:
+            if regex.search(link.text) or regex.search(link.url):
+                links.append(link)
+        if len(links) > 0:
```
Comment (Cito) [inline twill/utils.py]: Is there a reason why you don't simply return an empty list in that case?

File: twill/utils.py [hunk 2]
```diff
@@ -128,6 +128,17 @@ def find_link(self, pattern: str) -> Optional[Link]:
             if regex.search(link.text) or regex.search(link.url):
                 return link
         return None
+    
+    def find_links(self, pattern: str) -> Optional[List[Link]]:
+        """Find all links with a given pattern on the result page."""
+        regex = re.compile(pattern)
+        links = []
+        for link in self.links:
```
Comment (Cito) [inline twill/utils.py]: A list comprehension would be easier here.

**Review by Cito — CHANGES_REQUESTED** [review]
Thanks for the PR. I can add it if you think it's useful. See below for two comments.

File: twill/utils.py (hunk 1 above)
Comment (aruznieto) [reply twill/utils.py]: Fixed

File: twill/utils.py (hunk 2 above)
Comment (aruznieto) [reply twill/utils.py]: Fixed

File: twill/utils.py [hunk 3]
```diff
@@ -128,6 +128,12 @@ def find_link(self, pattern: str) -> Optional[Link]:
             if regex.search(link.text) or regex.search(link.url):
                 return link
         return None
+    
+    def find_links(self, pattern: str) -> Optional[List[Link]]:
```
Comment (Cito) [inline twill/utils.py]:
```suggestion
    def find_links(self, pattern: str) -> List[Link]:
```

The Optional is now not needed any more.

File: twill/utils.py [hunk 4]
```diff
@@ -128,6 +128,12 @@ def find_link(self, pattern: str) -> Optional[Link]:
...
+        """Find all links with a given pattern on the result page."""
+        regex = re.compile(pattern)
+        links = [link for link in self.links if regex.search(link.text) or regex.search(link.url)]
+        return links
```
Comment (Cito) [inline twill/utils.py]:
```suggestion
        return [link for link in self.links if regex.search(link.text) or regex.search(link.url)]
```

**Review by Cito — CHANGES_REQUESTED** [review]
Thanks for the fixes. Just two more small suggestions.

I will then merge and add unit tests for find_link and find_links afterwards.

**Review by Cito — APPROVED** [review]

**Pull request merged**